### PR TITLE
convert the backend to go

### PR DESCRIPTION
**Pull request by zpratt**
I'm doing this for a few reasons:

1. most of the Kubernetes community uses go. I want people to contribute to this
1. the client-go fake API is easily the best fake API for testing
   * followed by the [fabric8 fake](https://github.com/fabric8io/kubernetes-client/tree/master/kubernetes-server-mock)
   * I decided I'd rather learn go than either rewrite the backend in java or introduce an odd dependency just for a fake backend api

I've also made a design decision (tentatively): rather than building my own UI for this, it would be better to expose a `/metrics` endpoint that prometheus can scrape, which will return custom DORA-based metrics (like the number of deployments for an app and their timing). By doing this, we can then use a backend for prometheus that Grafana understands, which will allow for a more robust dashboard than what I'd ever have the time to create myself.

The eventual design will be a `/metrics` endpoint and then other handlers that implement API interfaces for Kubernetes webhooks so that we can track deployment frequency, etc. I'll convert the UI to an example application so that I can do E2E testing of this.

### CODE REVIEW

**Comment by zpratt** [comment]
My aim for this PR is to replace the current functionality in the node.js api and ensure the existing FE works with the rewritten API. More dramatic steps will be made afterwards

TODO:

* [ ] Nest `/` as `/api`, which will return a list of namespaces
* [ ] Implement `/api/{namespace}/pods`
* [ ] Update the UI to use the new response body structure, since I'm using a slice of structs for the namespace list instead of just returning an array of strings.

**Comment by zpratt** [comment]
@brigonzalez check this out when you have a sec

File: main.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	if err != nil {
+		klog.Errorf("error creating kubernetes client")
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func main() {
+	http.HandleFunc("/", apiRoot)
```
Comment (jones2026) [inline main.go:49]: Wouldn't this be `/api` to match the old API or are you hoping to add routing here so you don't need to add this to each handler?

**Comment by brigonzalez** [comment]
@zpratt very nice. I hadn't thought of using Prometheus as a source of data for DORA metrics. I can certainly make use of this data

File: main.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	json.NewEncoder(w).Encode(namespaceList)
+}
+
+func health(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte("pong"))
+}
+
+func getPods(w http.ResponseWriter, r *http.Request) {
+	pathParts := strings.Split(r.URL.Path, "/")[1:]
+	namespace := pathParts[len(pathParts) - 1]
```
Comment (zpratt) [inline main.go:32]: I'm intentionally deferring pulling in a router like Chi or mux, so I'm using this primitive approach for now.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	client, _ := createClient()
+
+	podList := pods.ListPods(client, namespace)
+
+	json.NewEncoder(w).Encode(podList)
+}
+
+// NOTE: i should use an environment variable to determine if i should start this with the fake server
```
Comment (zpratt) [inline main.go:40]: I think I'll do this after I figure out how to write tests for this file.

File: internal/pods/pods_test.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"log"
+	"testing"
+)
+
+func TestListPodsInNamespace(t *testing.T) {
```
Comment (zpratt) [inline internal/pods/pods_test.go:11]: I need to include a test that covers serializing the struct to json.

**Comment by zpratt** [comment]
This verified to be working, albeit from a manual testing perspective. This seeds a TDD approach for interacting with the Kubernetes API internally. Future PRs will continue to remove the need for manual testing, though I will have to do some research for:

1. how to test handlers that depend on a k8s client that needs to vary between tests and production. it seems there is a need for a factory interface and/or some sort of DI. I'm going to research this next.
1. how to do acceptance testing with prometheus in place to validate that scraping works

**Pull request merged**